unidist/core/backends/mpi/core/controller/garbage_collector.py

```diff
@@ -141,8 +137,6 @@ def regular_cleanup(self):
                     )
                     if executed_task_counter == self._task_counter:
                         self._send_cleanup_request(self._cleanup_list)
-                        # Clear the remaining references
-                        self._local_store.clear(self._cleanup_list)
```

Comment:
Why removed?

Reply:
Currently, local_store has a weak-dictionaries, which will be cleared themself. The only _data_id_map is used by workers and should be cleared only from CLEANUP operation.
Moreover `self._cleanup_list` is list of tuple and we need to recreate a MpiDataID intances from them